src/state-manager/TransactionQueue.ts

```diff
@@ -2559,8 +2628,21 @@ class TransactionQueue {
       )
     }
     if (queueEntry.collectedData[data.accountId] != null) {
-      // we have already collected this data
-      return
+      if (configContext.stateManager.collectedDataFix) {
+        // compare the timestamps and keep the newest
+        const existingData = queueEntry.collectedData[data.accountId]
+        if (data.timestamp > existingData.timestamp) {
```

Comment:
this will not work if coveredByUs is enabled

Reply:
I am not certain on the correct outcome here. there was another recent fix in this function to make sure that we don't over count the collected data.

Raising to blue team.

Reply:
This should still work fine even if `rejectSharedDataIfCovered: false` (i.e. we allow other nodes to send us the account that we cover), we should check the timestamp of incoming account data against the one we already collected. We should replace the collected data only if the incoming data is newer than the one we have.